design/jira-traceability.md

```diff
@@ -0,0 +1,150 @@
+# Traceability in unified pipelines - JIRA
+
+<p>As adoption of DevOps continues to accelerate, companies are innovating faster than ever- quickly adapting to changing customer needs and releasing at a much higher frequency. This coupled with the usage of myriad tools to plan, develop, build, deploy makes it quite cumbersome to keep a track of  the changes that are in the pipeline or already being used by the end customer.</p> 
+
+<p> Traceability is the ability to easily follow a change across all the systems involved in the development and deployment of a software. The full potential of traceability can only be realized when all systems involved in the lifecycle of a change seamlessly exchange data and let each user persona see a coherent, cross system information while staying in the system they interact with most. <br/> In this spirit, Azure Pipelines pulls information from supported planning systems and code repositories to show a list of work items and commits that were consumed in a pipeline run. While this information is vital for customers that use pipelines primarily, we also need to push pipeline information to external sytems so that other personas can have similar rich experience. </p>  
+
+<p> In this proposal we will discuss how we can enable this experience for users that track work items/issues in JIRA and build/deploy using Azure Pipelines</p>  
+
+## Requirements:
+
+The following information is required to light up traceability for JIRA: 
+
+1. Service connection of the source repository which contains the commits of the resource. This lets us parse commit messages and obtain the JIRA issues that have been linked to commits. 
+2. Service connection of the Jira server which is used to plan and track the changes done in a resource. This lets us talk to JIRA server and pull/push information.
+3. JIRA specific concept mappings like [Environment Type](https://confluence.atlassian.com/bitbucket/set-up-bitbucket-deployments-968683907.html). Jira uses this  categorization to progressively expose information to users in its UI. Since azure pipeline does not have such a concept, we need to take this mapping input from the user.  
+4. Filters (E.g. branch filters) to restrict the information transfer to JIRA.
+
+
+## Interaction with Jira Cloud 
+
+### Pull:
+ For pulling information from JIRA we would just need the service connection of the JIRA server. Using this we can pull details of consumed work items and display the list in  azure pipelines. 
```

Comment:
is this means, whenever we are showing linked workitems from traceability perspective, simply use this JIRA integration and show the data irrespective of the filters mentioned in the integration?

Reply:
yes